src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -298,29 +391,46 @@ private static Solution GetSolutionWithFormattedInterfaceDocument(Document unfor
             // extracted members are also not updated.
             if (typeToExtractFrom.TypeKind == TypeKind.Interface)
             {
-                return solutionWithFormattedInterfaceDocument;
+                return solution;
             }
 
-            var formattedSolution = GetSolutionWithUpdatedOriginalType(
-                solutionWithFormattedInterfaceDocument,
+            var unformattedSolution = solution;
+            foreach (var documentId in documentIds)
+            {
+                var document = solution.GetDocument(documentId);
+                var currentRoot = document.GetSyntaxRootSynchronously(cancellationToken);
```

Comment:
this should change to async version. are you trying to rely on the implementation detail that things are updated with PreservationMode.PreserveIdentity? if you do, you should add assert or something somewhere so that you make sure that nobody accidently update tree/solution with PreservationMode.PreserveValue

Reply:
or if you are 100% sure, use TryGetXXX method such as TryGetRoot to show that you believe root must be already in memory.